eis_toolkit/utilities/file_io.py

```diff
@@ -0,0 +1,112 @@
+from pathlib import Path
+from typing import List, Optional
+
+import fiona
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException, UnsupportedFileExtensionException
+
+
+def get_supported_raster_extensions() -> List[str]:
+    """Get raster file extensions supported by rasterio."""
+    extensions = list(rasterio.drivers.raster_driver_extensions().keys())
+    return extensions
+
+
+def get_supported_vector_extensions() -> List[str]:
+    """Get vector file extensions supported."""
+    extensions = ["shp", "geojson", "gpkg", "csv", "kml", "gml"]
+    return extensions
+
+
+def read_spatial(file_path: Path) -> rasterio.io.DatasetReader | gpd.GeoDataFrame:
+    """Read and open a vector or raster file."""
+    file_extension = file_path.split(".")[-1]
+    if file_extension in get_supported_vector_extensions:
+        data = rasterio.open(file_path)
+    elif file_extension in get_supported_vector_extensions():
+        data = gpd.read_file(file_path)
+    else:
+        raise UnsupportedFileExtensionException(
+            f"File extension {file_extension} is not supported. Provide a raster or vector file."
+        )
+    return data
```

Comment:
Good points @nialov . I'll think about this a little bit. I can imagine it was thought in the beginning that classes won't be needed, but we are not bound by the initial decisions as now we know more about the toolkit and our options.

Reply:
If I'm not mistaken, `list(rasterio.drivers.raster_driver_extensions().keys())` should cover all formats supported by rasterio, but a similar listing does not exist for vector data. Did you have something like this as an alternative in your mind Nikolas?

```
def read_spatial(file_path: Path):
    try:
        data = read_raster(file_path)
    except rasterio.errors.RasterioIOError:
        try:
            data = read_vector(file_path)
        except (ValueError, OSError):
            try:
                data = read_csv(file_path)
            pd.errors.ParserError:
                raise FileReadError("Unsupported file format or corrupted file")

    return data
```

Here I'm not sure if all the possible relevant exceptions are caught.

Reply:
Yes, something like that. It will probably require some trial and error to catch all exceptions. However, just as a side note, I would not consider this functionality of accepting tabular and geometry data in place of rasters a priority in terms of time. We should focus on making everything work with rasters first and then start adding the additional formats.